rio/test/RIO/VectorSpec.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+module RIO.VectorSpec where
+
+import Test.Hspec
+import RIO
+import qualified RIO.Vector.Boxed as VB
+
+spec :: Spec
+spec =
```

Comment:
Some quickchecks/property tests would be a great addition here.

Reply:
Ah, I didn't write any because I didn't see any in `ListSpec.hs`.

Reply:
I was about to replace all the tests with: 

```hs
spec :: Spec
spec =
  describe "sliceMaybe" $ do
    prop "is consistent with `slice` (pathological cases)" $
      \(QC.Large i) (QC.Large n) v
        -> sliceProp i n (DV.fromList v)
    prop "is consistent with `slice` (less pathological cases)" $
      \(QC.NonNegative i) (QC.NonNegative n) (QC.NonEmpty v)
        -> sliceProp i n (DV.fromList v)
  where
    sliceProp :: Int -> Int -> DV.Vector Char -> QC.Property
    sliceProp i n v = QC.withMaxSuccess 10000 $ case VB.sliceMaybe i n v of
      Just v' -> DV.slice i n v `shouldBe` v'
      Nothing -> evaluate (DV.slice i n v) `shouldThrow` anyException
```

(The second property could be skipped since it is implicit in the first and the number of cases could be reduced to the default of 100, but the coverage may not be extensive.)

However, these tests resulted in:

```
rio        > RIO.Vector       
rio        >   sliceMaybe     
rio        >     is consistent with `slice` (pathological cases) FAILED [1]
rio        >     is consistent with `slice` (less pathological cases)
rio        >       +++ OK, passed 10000 tests.
rio        >                  
rio        > Failures:        
rio        >                  
rio        >   test/RIO/VectorSpec.hs:23:18: 
rio        >   1) RIO.Vector.sliceMaybe is consistent with `slice` (pathological cases)
rio        >        Falsifiable (after 82 tests and 25 shrinks):
rio        >          Large {getLarge = 2104973872246037021}
rio        >          Large {getLarge = 7118398164608738787}
rio        >          ""      
rio        >        did not get expected exception: SomeException
```

Further investigation led me to open <https://gitlab.haskell.org/ghc/ghc/issues/17233>.

Reply:
What makes you think this is a GHC bug instead of a `vector` library bug?

Reply:
I just used the heuristic that if it is a segfault or the error message says Please report this as a GHC bug it should be reported here. But it may well be (probably is?) a vector bug so I reported it as https://github.com/haskell/vector/issues/257 too.